### PR TITLE
Adding metric group name as dimension to yammer reporter

### PR DESCRIPTION
**Pull request by keitwb**

### CODE REVIEW

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java [hunk 1]
```diff
@@ -241,6 +241,11 @@ void addMetric(Metric metric, MetricName codahaleName,
             }
         }
 
+        if (codahaleName.getGroup() != null && codahaleName.getGroup() != "") {
```
Comment (tedoc2000) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: if (!StringUtils.isEmpty(codahaleName.getGroup()) {

**Comment by keitwb** [comment]
@mpetazzoni This is needed because we have a customer who has an existing application that reports metrics using yammer that reuses metric names across groups.  They want to be able to easily distinguish those metrics in the app without having to change the yammer code in their application.

**Comment by keitwb** [comment]
@tedoc2000 does this mean going to 0.1.0 since our current version is 0.0.36?

**Comment by tedoc2000** [comment]
@keitwb It would be a major change in functionality so I would say 1.0.0

**Comment by mpetazzoni** [comment]
Are metric groups a thing in codahale too? Are we going to need this in `signalfx-codahale`? If we do, how many other customers is this going to impact (in addition to us)?

**Comment by keitwb** [comment]
@mpetazzoni From what I can tell from http://metrics.dropwizard.io/3.1.0/manual/core/ it doesn't look like newer versions of Codahale has the concept of metric group names, but I'm not terribly familiar with this library, so perhaps @tedoc2000 knows.

File: signalfx-yammer/pom.xml [hunk 2]
```diff
@@ -57,6 +57,10 @@
       <artifactId>guava</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.apache.commons</groupId>
```
Comment (tedoc2000) [inline signalfx-yammer/pom.xml]: There is an equivalent of StringUtils.isEmpty in guava.. Lets use that one and not add yet another dependency

**Comment by tedoc2000** [comment]
@keitwb don't know anything about the metric_group thing, by my googling matches yours so it seems like a yammer thing only

**Comment by mpetazzoni** [comment]
LGTM. @keitwb if you have some cycles can you also figure out what's going on with the TravisCI build?

**Pull request merged**